sound/soc/sof/pcm.c

```diff
@@ -619,6 +619,27 @@ static int sof_pcm_new(struct snd_soc_component *component,
 	return 0;
 }
 
+static void ssp_dai_config_pcm_params_match(struct snd_sof_dev *sdev, const char* name,
+					    struct snd_pcm_hw_params *params)
+{
+	struct sof_ipc_dai_config *config;
+	struct snd_sof_dai *dai;
+	int i;
+
+	/* we might have several dai's associated with same link, need to loop all */
```

Comment:
this part is not very clear too, why all DAI's, shouldnt we check for direction too?

Reply:
So I understood this is for all dai's in a link. So actually the calling function call sof_find_dai which finds only SSP_IN.
If you can come up how I can dig up the dir from both dai and pcm hw params, I can do that

Reply:
hmm ok for SSP, I guess the config is the same for both playback and cpature DAI. So instead of saying several, can we change to say both playback/capture DAIs?

Reply:
ok. I tried to make the comment better and I renamed the parameter to link_name to make it more clear what is happening here...